### PR TITLE
feat: onboard reddit cloud mode destination

### PR DESCRIPTION
**Pull request by ujjwal-ab**
## Description of the change
Resolves INT-910
> Onboard reddit cloud mode destination

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/cdk/v2/destinations/reddit/procWorkflow.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            "conversion_id": .properties.conversionId || .messageId,
+          });
+          $.removeUndefinedAndNullValues(customFields)
+
+      - name: eventType
+        template: |
+          let event = .message.event;
+          let eventInLowerCase = event.toLowerCase();
+          let eventNames = .destination.Config.eventsMapping.(){.from === event}.to[] ?? [];
+          eventNames.length === 0 ? eventNames = $.ecomEventMaps.(){eventInLowerCase in .src}.dest[] ?? [];
```
Comment (koladilip) [inline src/cdk/v2/destinations/reddit/procWorkflow.yaml:61]: eventNames.length === 0 ? eventNames = $.ecomEventMaps.(){eventInLowerCase in .src}.dest[] : [];
?? -> :

**Review by yashasvibajpai — APPROVED** [review]

File: test/integrations/destinations/reddit/processor/data.ts [hunk 2]
```diff
@@ -0,0 +1,1519 @@
...
+                      event_at: '2019-10-14T09:03:17.562Z',
+                      event_type: {
+                        tracking_type: 'Purchase',
+                      },
+                      user: {
+                        email: 'ac144532d9e4efeab19475d9253a879173ea12a3d2238d1cb8a332a7b3a105f2',
+                        external_id:
+                          '7b023241a3132b792a5a33915a5afb3133cbb1e13d72879689bf6504de3b036d',
+                        ip_address:
+                          'e80bd55a3834b7c2a34ade23c7ecb54d2a49838227080f50716151e765a619db',
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply test/integrations/destinations/reddit/processor/data.ts:115]: I think config field is not needed as hashing is suggested for only 2-3 fields and we will hash it be default.

[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply test/integrations/destinations/reddit/processor/data.ts:115]: But, hashing is suggested by reddit for 2-3 fields only, so should we expect that customer will know that and will send hashed data for those fields only?

File: src/cdk/v2/destinations/reddit/utils.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+          response.destination = event.destination;
+          response.metadata = [event.metadata];
+        } else {
+          response.message.body.JSON.events.push(...event.message[0].body.JSON.events);
+          response.metadata.push(event.metadata);
+        }
+      });
+      batchedEvents.push(response);
+    });
+  }
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply src/cdk/v2/destinations/reddit/utils.js:23]: Updated

File: src/cdk/v2/destinations/reddit/procWorkflow.yaml [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            "screen_dimensions": {"width": .context.screen.width, "height":  .context.screen.height},
+          });
+          $.removeUndefinedAndNullValues(user)
+
+      - name: customFields
+        template: |
+          const customFields = .message.().({
+            "currency": .properties.currency,
+              "value": .properties.revenue !== undefined ? Number(.properties.revenue) : undefined,
+            "item_count": .properties.itemCount && Number(.properties.numOfItems),
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply src/cdk/v2/destinations/reddit/procWorkflow.yaml:54]: Updated

File: test/integrations/destinations/reddit/router/data.ts [hunk 5]
```diff
@@ -0,0 +1,501 @@
...
+                },
+              },
+            },
+          ],
+        },
+      },
+    },
+  },
+  {
+    name: 'ortto',
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply test/integrations/destinations/reddit/router/data.ts:374]: Updated

File: test/integrations/destinations/reddit/processor/data.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply test/integrations/destinations/reddit/processor/data.ts:123]: Added a field in config

**Review by yashasvibajpai — DISMISSED** [review]

File: src/v0/destinations/reddit/networkHandler.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+const {
+  getDynamicErrorType,
+  processAxiosResponse,
+} = require('../../../adapters/utils/networkUtils');
+
+const redditRespHandler = (destResponse) => {
+  const { status, response } = destResponse;
+
+  // to handle the case when authorization-token is invalid
+  if (status === 401) {
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply src/v0/destinations/reddit/networkHandler.js:15]: No error doc, I could find. Not sure about other errors we can get.

[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply src/v0/destinations/reddit/networkHandler.js:15]: Could generate only this.

**Review by yashasvibajpai — DISMISSED** [review]

**Review by ItsSudip — DISMISSED** [review]

**Review by sanpj2292 — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**